src/components/absorber/carbon_handler.cairo

```diff
@@ -227,6 +239,35 @@ mod AbsorberComponent {
             };
             cc_distribution.span()
         }
+
+        fn rebase_vintage(
+            ref self: ComponentState<TContractState>, token_id: u256, new_cc_supply: u64
+        ) {
+            let mut stored_vintages: List<CarbonVintage> = self.Absorber_vintage_cc.read();
+            let mut index = 0;
+            loop {
```

Comment:
We might want to use a hashmap here instead of List to avoid looping.
It might imply not having token_ids being years after all.

Reply:
#57